.github/workflows/matlab.yml

```diff
@@ -1,20 +1,29 @@
 name: MATLAB Tests and Packaging
 on: [push]
 
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref == 'refs/heads/main' && github.run_id || github.event.pull_request.number || github.ref }}
+  cancel-in-progress: true
+
 jobs:
 
   test:
     name: Run MATLAB Tests
-    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        os: [ubuntu-latest, windows-latest, macos-latest]
+        release: ["R2023b", "R2024a"]
```

Comment:
This is cool, will definitely be stealing this for the flow with the different versions. It looks like from your intergrated tests the test time has approximately doubled form when you added this. Maybe you could get rid of Mac Os as we never seem to use it, but up to you. And the test time is not that long anyway ~6mins

Reply:
Yes, I've had a think, and until proven otherwise, I think it's ok to run on Linux only. It's the fastest platform and MATLAB _should_ be platform agnostic.